### DISCUSSION TITLE
Image optimisation at build time

### DISCUSSION THREAD

**Discussion by vitalibakhur** [vercel/next.js#51865] [Help]
### Summary

Is it possible to optimize images at build time without different plugins like this https://github.com/dc7290/next-export-optimize-images . This plugin not ok for me. On my opinion the best choice its create cache image on build time. but how to do it

**Comment by rasif-sahl** [comment]
Yes, it is possible to optimize images at build time in Next.js without relying on external plugins. Next.js provides an Image component that automatically optimizes images for you during the build process.

Here's how you can optimize images at build time using Next.js:

1. Create a folder named public in the root directory of your Next.js project if it doesn't already exist.

2. Place your images in the public folder. For example, if you have an image named my-image.jpg, you can place it in public/my-image.jpg.

3. In your code, use the Image component from next/image to display the image. Provide the path to the image using the src attribute of the Image component.

```
import Image from 'next/image';

function MyComponent() {
  return (
    <div>
      <Image
        src="/my-image.jpg"
        alt="My Image"
        width={500}
        height={300}
      />
    </div>
  );
}

export default MyComponent;
```

In the example above, the Image component from next/image is used to display the image. The src attribute is set to the path of the image file in the public folder (/my-image.jpg). You can also specify the width and height attributes to control the size of the image.

During the build process, Next.js will automatically optimize the image for you, creating different versions of the image with multiple resolutions. It will generate a unique URL for each optimized version of the image.

When the page is loaded, Next.js will serve the appropriate version of the image based on the device's screen size, providing an optimized image for the user.

By using the Image component provided by Next.js, you can easily optimize images at build time without the need for additional plugins or external dependencies.

**Reply by vitalibakhur** [reply]
@rasif-sahl  hm, i think it doesn't work so, because now i have the same logic and npm run build don't create this images. when you go to the page with images, only after this next js create cache images. And you can see this images on folder .next

**Comment by leerob** [comment]
Closing as a duplicate of https://github.com/vercel/next.js/discussions/19065.
